src/DataSource.ts

```diff
@@ -174,27 +184,24 @@ export class DataSource extends DataSourceApi<AkenzaQuery, AkenzaDataSourceConfi
                     resolve(response.data.content[0]);
                 },
                 (error: FetchError) => {
-                    reject(this.generateErrorMessage(error));
+                    reject(DataSource.generateErrorMessage(error));
                 }
             );
         });
     }
 
     private executeRequest<T>(path: string, method: string, params?: any, data?: any): Observable<FetchResponse<T>> {
         const options: BackendSrvRequest = {
-            url: buildUrl(this.config.baseUrl, { path }),
+            url: buildUrl(this.url!, { path }),
```

Comment:
what does the ! do?

Reply:
it indicates that an optional property is not optional at this point anymore (as the argument is non-nullable for example). I removed this one as it's no longer necessary but still needed in the constructor where the base url is built ^^